leader-election-impl/src/main/java/com/palantir/paxos/SingleLeaderPinger.java

```diff
@@ -42,13 +43,13 @@
 
     private static final Logger log = LoggerFactory.getLogger(SingleLeaderPinger.class);
 
-    private final ConcurrentMap<UUID, PingableLeader> uuidToServiceCache = Maps.newConcurrentMap();
-    private final Map<PingableLeader, ExecutorService> leaderPingExecutors;
+    private final ConcurrentMap<UUID, LeaderPingerContext<PingableLeader>> uuidToServiceCache = Maps.newConcurrentMap();
```

Comment:
Curious why this isn't a cache too? It doesn't look like anything ever expires values from here?

Reply:
no reason for it not to be, just been like that since whenever really